### PR TITLE
Add LLVM module support for fcmp instruction

### LINKED ISSUE #311: Add LLVM module support for `fcmp` instruction
The condition codes we need are ugt/uge/ult/ule, but we might as well add support for all of them.

Needed for #262.

### PR DESCRIPTION
**Pull request by heshanpadmasiri**
## Purpose
Add LLVM module support for `fcmp` instruction
Resolves #311

## Automation tests
 - Unit tests 
   > Code coverage information
 - Integration tests
   > Details about the test cases and coverage

## Security checks
 - Followed secure coding standards in http://wso2.com/technical-reports/wso2-secure-engineering-guidelines? yes/no
 - Ran FindSecurityBugs plugin and verified report? yes/no
 - Confirmed that this PR doesn't commit any keys, passwords, tokens, usernames, or other secrets? yes/no

### CODE REVIEW

File: compiler/modules/print.llvm/common.bal [hunk 1]
```diff
@@ -85,6 +85,8 @@ type BinaryOp IntOp|FloatArithmeticOp;
 // Corresponds to LLVMIntPredicate
 public type IntPredicate "eq"|"ne"|"ugt"|"uge"|"ult"|"ule"|"sgt"|"sge"|"slt"|"sle";
 
+// Corresponds to LLVMRealPredicate
+public type RealPredicate "false"|"oeq"|"ogt"|"oge"|"olt"|"ole"|"one"|"ord"|"ueq"|"ugt"|"uge"|"ult"|"ule"|"une"|"uno"|"true";
```
Comment (manuranga) [inline compiler/modules/print.llvm/common.bal]: I am renaming `RealType` to `BinaryFp` as suggested in [#306 (comment)](https://github.com/ballerina-platform/nballerina/pull/306#discussion_r688860125).
Let's change this to `BinaryFpPredicate` to match.

Comment (jclark) [reply compiler/modules/print.llvm/common.bal]: Our LLVM module is using terminology from the LLVM lang ref (in preference to LLVM API) and the lang ref calls these floating point types https://llvm.org/docs/LangRef.html#floating-point-types, so I think maybe FpType or FloatingPointType is better.

Comment (jclark) [reply compiler/modules/print.llvm/common.bal]: Binary is going to get confused with Binary operators.

Comment (jclark) [reply compiler/modules/print.llvm/common.bal]: Looking at the code I think `FloatType` will work best.

Comment (heshanpadmasiri) [reply compiler/modules/print.llvm/common.bal]: Renamed to `FloatPredicate` to match `FloatType`

**Pull request merged**